be/src/exec/pipeline/exchange/local_exchange.cpp

```diff
@@ -174,6 +162,130 @@ Status PartitionExchanger::accept(const ChunkPtr& chunk, const int32_t sink_driv
     return Status::OK();
 }
 
+OrderedPartitionExchanger::OrderedPartitionExchanger(const std::shared_ptr<ChunkBufferMemoryManager>& memory_manager,
+                                                     LocalExchangeSourceOperatorFactory* source,
+                                                     std::vector<ExprContext*> partition_expr_ctxs)
+        : LocalExchanger("OrderedPartition", memory_manager, source),
+          _partition_exprs(std::move(partition_expr_ctxs)) {}
+
+Status OrderedPartitionExchanger::prepare(RuntimeState* state) {
+    RETURN_IF_ERROR(LocalExchanger::prepare(state));
+    RETURN_IF_ERROR(Expr::prepare(_partition_exprs, state));
+    RETURN_IF_ERROR(Expr::open(_partition_exprs, state));
+    _channel_row_nums.resize(source_dop());
+    _channel_row_nums.assign(source_dop(), 0);
+    return Status::OK();
+}
+
+void OrderedPartitionExchanger::close(RuntimeState* state) {
+    Expr::close(_partition_exprs, state);
+    LocalExchanger::close(state);
+}
+
+Status OrderedPartitionExchanger::accept(const ChunkPtr& chunk, const int32_t sink_driver_sequence) {
+    DCHECK_EQ(sink_driver_sequence, 0);
+
+    Columns partition_columns(_partition_exprs.size());
+    for (size_t i = 0; i < partition_columns.size(); ++i) {
+        ASSIGN_OR_RETURN(partition_columns[i], _partition_exprs[i]->evaluate(chunk.get()));
+        DCHECK(partition_columns[i] != nullptr);
+    }
+
+    ChunkPtr output_chunk = chunk;
+
+    size_t channel_id;
+    size_t min_channel_id = _find_min_channel_id();
+    if (_previous_chunk == nullptr || _previous_channel_id == min_channel_id) {
+        channel_id = min_channel_id;
+    } else {
+        // Check if the joint of two consecutive chunks are the same
+        bool is_joint_equal = true;
+        for (size_t i = 0; i < partition_columns.size(); ++i) {
```

Comment:
suggest to write util function with the signature as follows

```cpp
compare_at(const Columns& lhs, size_t lhs_pos, const Columns& rhs, size_t rhs_pos);
```

Reply:
updated